node/libs/roles/src/validator/messages/block.rs

```diff
@@ -4,6 +4,56 @@ use super::CommitQC;
 use crypto::{sha256, ByteFmt, Text, TextFmt};
 use std::fmt;
 
+/// Version of the consensus algorithm used by the leader in a given view.
+/// Currently it is stored in the BlockHeader (TODO: consider whether it should be present in every
+/// consensus message).
+/// Currently it also determines the format and semantics of the block payload
+/// (TODO: consider whether the payload should be versioned separately).
```

Comment:
I had some afterthoughts after that discussion in the context of the above: if we keep the version (only) in the block, then some messages don't include the version at all. And if we (hypothetically) added version to both header and the messages, then forcing them to be the same wouldn't bring any benefit and might be constraining. Not a strong opinion, just writing down my thoughts.

Reply:
Well, if we include the version in every message, then we don't need to include it in the block header. The block header is never sent by itself over the wire, either it is sent in a leader prepare or in a final block (which would contain the version in the commit QC). It would even make some sense since the protocol version isn't technically a property of the block, but of the nodes.